contributors/design-proposals/sig-storage/volume-topology-scheduling.md

```diff
@@ -0,0 +1,580 @@
+# Volume Topology-aware Scheduling
+
+Authors: @msau42
+
+This document presents a detailed design for making the default Kubernetes
+scheduler aware of volume topology constraints, and making the
+PersistentVolumeClaim (PVC) binding aware of scheduling decisions.
+
+
+## Goals
+* Allow a Pod to request one or more topology-constrained Persistent
+Volumes (PV) that are compatible with the Pod's other scheduling
+constraints, such as resource requirements and affinity/anti-affinity
+policies.
+* Support arbitrary PV topology constraints (i.e. node,
+rack, zone, foo, bar).
+* Support topology constraints for statically created PVs and dynamically
+provisioned PVs.
+* No scheduling latency performance regression for Pods that do not use
+topology-constrained PVs.
+
+
+## Non Goals
+* Fitting a pod after the initial PVC binding has been completed.
+    * The more constraints you add to your pod, the less flexible it becomes
+in terms of placement.  Because of this, tightly constrained storage, such as
+local storage, is only recommended for specific use cases, and the pods should
+have high priority.
+    * Pod priorities could handle this scenario.  PV taints/tolerations could
+also handle this case.
+* Binding decision considering scheduling constraints from two or more pods
+sharing the same PVC.
+    * The scheduler itself only handles one pod at a time.  It’s possible the
+two pods may not run at the same time either, so there’s no guarantee that you
+will know both pod’s requirements at once.
+    * For two+ pods simultaneously sharing a PVC, this scenario may require an
+operator to schedule them together.
+    * For two+ pods non-simultaneously sharing a PVC, this scenario could be
+handled by pod priorities.
+
+
+## Problem
+Volumes can have topology constraints that restrict the set of nodes that the
+volume is accessible from.  For example, a GCE PD can only be accessed from a
```

Comment:
/the volume is accessible from./can mount them./

alternatively /can access them./